src/main/java/org/dcsa/bkg/service/impl/BookingSummaryServiceImpl.java

```diff
@@ -23,18 +23,20 @@
 
   @Override
   protected Flux<BookingSummaryTO> bulkMapDM2TO(Flux<Booking> bookingFlux) {
-    return bookingFlux.concatMap(
-        booking ->
-            vesselRepository
-                .findByIdOrEmpty(booking.getVesselId())
-                .map(
-                    vessel -> {
-                      BookingSummaryTO bookingSummaryTO =
-                          mapper.bookingSummaryTOFromBooking(booking);
-                      bookingSummaryTO.setVesselIMONumber(vessel.getVesselIMONumber());
-                      return bookingSummaryTO;
-                    })
-                .defaultIfEmpty(mapper.bookingSummaryTOFromBooking(booking)));
+    return bookingFlux
+        .flatMap(booking -> booking.getValidUntil() == null ? Mono.just(booking) : Mono.empty())
```

Comment:
it seems like this could be done better with a .filter() to make it more idiomatic reactor